ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,299 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <cctype>
+#include <ngraph/validation_util.hpp>
+#include <string>
+#include <unordered_map>
+
+#include "itt.hpp"
+#include "ngraph/op/einsum.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v7::Einsum, "Einsum", 7);
+
+op::v7::Einsum::Einsum(const OutputVector& inputs, const std::string& equation)
+    : Op(inputs)
+    , m_equation(equation)
+{
+    // normalize input equation by removing extra white-spaces from the equation
+    m_equation.erase(std::remove_if(m_equation.begin(), m_equation.end(), ::isspace),
+                     m_equation.end());
+
+    constructor_validate_and_infer_types();
+}
+
+bool op::v7::Einsum::is_subscript_correct(const std::string& subscript, bool& is_ellipsis_met)
```

Comment:
We shouldn't add this method as a part of an operation, it can be a local function. I don't see any reasons why it should be a part of operation.

Reply:
Done